src/vectorise.F90

```diff
@@ -64,78 +109,108 @@ PROGRAM main
+        END DO
+
+        ! Allocate array and populate it ...
+        CALL sub_allocate_array(y, "y", nyScaled + 1_INT64, .TRUE._INT8)
+        DO iy = 0_INT64, nyScaled
+            y(iy + 1_INT64) = 180.0e0_REAL64 * (0.5e0_REAL64 * REAL(nyScaled, kind = REAL64) - REAL(iy, kind = REAL64)) / REAL(nyScaled, kind = REAL64) ! [°]
+        END DO
+
         ! Loop over elevations ...
         ! NOTE: Rounded to the nearest integer, Mount Everest is 8,849m ASL.
         ! NOTE: The output of "downscale" reports that the highest pixel in the
         !       un-scaled dataset is 8,752m ASL.
         DO z = 250_INT16, 8750_INT16, 250_INT16
-            WRITE(fmt = '(" > Searching for elevation of ", i4, "m ...")', unit = OUTPUT_UNIT) z
+            ! Determine output file names ...
+            WRITE(fnameHDF, fmt = '("../data/scale=", i2.2, "km/elev=", i4.4, "m.h5")') scale, z
+            WRITE(fnamePGM, fmt = '("../data/scale=", i2.2, "km/elev=", i4.4, "m.pgm")') scale, z
+
+            ! Skip if the output file exists ...
+            INQUIRE(file = TRIM(fnameHDF), exist = fexist)
+            IF(fexist)THEN
+                WRITE(fmt = '("  Skipping for elevation of ", i4, "m (the output already exists).")', unit = OUTPUT_UNIT) z
+                FLUSH(unit = OUTPUT_UNIT)
+                CYCLE
+            END IF
+
+            WRITE(fmt = '("  Searching for elevation of ", i4, "m ...")', unit = OUTPUT_UNIT) z
             FLUSH(unit = OUTPUT_UNIT)
 
-            ! Determine directory name and make it ...
-            WRITE(dname, fmt = '("../data/scale=", i2.2, "km/elev=", i4.4, "m")') scale, z
-            CALL EXECUTE_COMMAND_LINE(                                          &
-                "mkdir -p " // TRIM(dname),                                     &
-                  cmdmsg = errmsg,                                              &
-                exitstat = errnum                                               &
+            ! Create HDF5 file ...
+            CALL H5FCREATE_F(                                                   &
+                access_flags = H5F_ACC_TRUNC_F,                                 &
+                        name = TRIM(fnameHDF),                                  &
+                     file_id = hUnit,                                           &
+                      hdferr = errnum                                           &
             )
             IF(errnum /= 0)THEN
-                WRITE(fmt = '("ERROR: ", a, ". ERRMSG = ", a, ". ERRNUM = ", i3, ".")', unit = ERROR_UNIT) "mkdir failed", TRIM(errmsg), errnum
+                WRITE(fmt = '("ERROR: ", a, ". ERRNUM = ", i3, ".")', unit = ERROR_UNIT) "H5FCREATE_F() failed", errnum
                 FLUSH(unit = ERROR_UNIT)
                 STOP
             END IF
 
-            ! Create short-hands ...
```

Comment:
This entire block on depended on `scale` rather than also `z`, so I have moved it out one more loop.